eventdisplay/FairXMLNode.cxx

```diff
@@ -8,232 +8,105 @@
-	if(fMode == kModeNull){
-		//FairCout::Error("dont call for null node");
-		return 0;
+FairXMLNode::FairXMLNode(const FairXMLNode& other):
+		FairXMLNode(other.GetName(),other.GetValue()){
+	for(int i=0;i<other.fChildren.GetEntries();i++){
+		fChildren.Add(new FairXMLNode(*other.GetChild(i)));
 	}
-	return fNChildren;
-}
-
-Int_t FairXMLNode::GetNAttributes() const {
-	if(fMode == kModeNull){
-		//FairCout::Error("not for null node");
-		return 0;
-	}
-	return fNAttrib;
-}
-
-TList* FairXMLNode::GetChildren() const {
-	if(fMode == kModeNull){
-		//FairCout::Error("no child in null node");
-		return NULL;
+	for(int i=0;i<other.fAttrib.GetEntries();i++){
+		fAttrib.Add(new FairXMLAttrib(*other.GetAttrib(i)));
 	}
-	return fChildren;
 }
 
-FairXMLNode * FairXMLNode::fgNullInstance = NULL;
-
 FairXMLNode* FairXMLNode::GetChild(TString name, Int_t count) const{
-	if(fMode == kModeNull){
-		//FairCout::Error("no child in null node");
-		return FairXMLNode::NullInstance();
-	}
-	if(fNChildren==0){
-		//FairCout::Error("No children in this node");
-		return FairXMLNode::NullInstance();
-	}
 	Int_t control_index = 0;
-	for(int i=0;i<fChildren->GetEntries();i++){
-		FairXMLNode *node = (FairXMLNode*)fChildren->At(i);
+	for(int i=0;i<fChildren.GetEntries();i++){
+		FairXMLNode *node = GetChild(i);
 		TString temp = node->GetName();
 		if(temp==name){
 			control_index++;
 		}
 		if(control_index>count)
 			return node;
 	}
-	//FairCout::Error("No child found by name");
-	return FairXMLNode::NullInstance();
-}
-
-FairXMLNode* FairXMLNode::GetChild(Int_t index) const{
-	if(fMode == kModeNull){
-		return FairXMLNode::NullInstance();
-	}
-	if(fNChildren>index){
-		return (FairXMLNode*)fChildren->At(index);
-	}else{
-		return FairXMLNode::NullInstance();
-	}
-}
-
-Bool_t FairXMLNode::IsImport() const{
-	if(fMode==kModeImport) return kTRUE;
-	return kFALSE;
-}
-
-Bool_t FairXMLNode::IsNull() const {
-	if(fMode == kModeNull) return kTRUE;
-	return kFALSE;
-}
-
-FairXMLNode* FairXMLNode::NullInstance() {
-	if(fgNullInstance){
-		return fgNullInstance;
-	}
-	fgNullInstance = new FairXMLNode();
-	return fgNullInstance;
-}
-
-FairXMLNode::FairXMLNode() {
-	fName = "null";
-	fValue= "null";
-	fChildren = NULL;
-	fAttrib = NULL;
-	fNChildren = 0;
-	fNAttrib = 0;
-	fMode = kModeNull;
+	return nullptr;
 }
 
-void FairXMLNode::SetValue(TString value) {
-	if(fMode==kModeExport){
-		fValue = value;
+FairXMLAttrib* FairXMLNode::GetAttrib(TString name) const{
+	Int_t control_index = 0;
+	for(int i=0;i<fAttrib.GetEntries();i++){
+		FairXMLAttrib *node = static_cast<FairXMLAttrib*>(fAttrib.At(i));
+		TString temp = node->GetName();
+		if(temp==name){
+			return node;
+		}
 	}
+	return nullptr;
 }
 
-void FairXMLNode::SetName(TString name) {
-	if(fMode==kModeExport){
-		fName = name;
+FairXMLNode* FairXMLNode::GetChild(Int_t index) const{
+	if(index<fChildren.GetEntries()){
+		return static_cast<FairXMLNode*>(fChildren.At(index));
 	}else{
+		return nullptr;
 	}
 }
```

Comment:
The boundary check [is not necessary](https://root.cern.ch/doc/master/classTList.html#a9468338df7118022e6743c5fa6ac668b) (and incomplete, what about negative arguments?), so `return static_cast<FairXMLNode*>(fChildren.At(index));` should suffice.